eventio/simtel/objects.py

```diff
@@ -1610,51 +1610,43 @@ def __str__(self):
         )
 
     def parse(self):
-        assert_exact_version(self, supported_version=2)
+        assert_max_version(self, 2)
+        version = self.header.version
         self.seek(0)
         byte_stream = BytesIO(self.read())
 
         event = self.header.id
         shower_num = read_int(byte_stream)
         n_tel = read_int(byte_stream)
         n_pes = read_array(byte_stream, 'i4', n_tel)
-        n_pixels = read_array(byte_stream, 'i4', n_tel)
-
-        # NOTE:
-        # I don't see how we can speed this up easily since the length
-        # of this thing is not known upfront.
-
-        # pixel_pe: a list (running over telescope_id)
-        #         of 2-tuples: (pixel_id, pe)
-        pixel_pe = []
-        mask = (n_pixels > 0) & (n_pes > 0)
 
-        for n_pe, n_pixels in zip(n_pes[mask], n_pixels[mask]):
-            non_empty = read_short(byte_stream)
-            pixel_id = read_array(byte_stream, 'i2', non_empty)
-            pe = read_array(byte_stream, 'i4', non_empty)
-            pixel_pe.append((pixel_id, pe))
-
-        photons = read_array(byte_stream, 'f4', n_tel)
-        photons_atm = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_3_6 = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_qe = read_array(byte_stream, 'f4', n_tel)
-        photons_atm_400 = read_array(byte_stream, 'f4', n_tel)
-
-        return {
+        result = {
             'event': event,
             'shower_num': shower_num,
             'n_tel': n_tel,
             'n_pe': n_pes,
-            'n_pixels': n_pixels,
-            'pixel_pe': pixel_pe,
-            'photons': photons,
-            'photons_atm': photons_atm,
-            'photons_atm_3_6': photons_atm_3_6,
-            'photons_atm_qe': photons_atm_qe,
-            'photons_atm_400': photons_atm_400,
         }
 
+        # this code was never used according to K. Bernlöhr
+        # so we raise an error in case this happens
+        n_pixels = read_array(byte_stream, 'i4', n_tel)
+        if np.count_nonzero(n_pixels) > 0:
+            raise NotImplementedError(
+                f'Reading pixel data in {self} is not supported, '
+                'please contact the pyeventio maintainers if this happens'
+            )
+
+        if version >= 1:
+            result['photons'] = read_array(byte_stream, 'f4', n_tel)
+            result['photons_atm'] = read_array(byte_stream, 'f4', n_tel)
+            result['photons_atm_3_6'] = read_array(byte_stream, 'f4', n_tel)
+            result['photons_atm_qe'] = read_array(byte_stream, 'f4', n_tel)
+
+        if version >= 2:
+            result['photons_atm_400'] = read_array(byte_stream, 'f4', n_tel)
+
+        return result
```

Comment:
Regarding usability:
`result` will have different keys depending on which version of this object we happened to get. 
So consumers of result might use a conditional like this:
```
if 'photons_atm_400' in result:
    # code dealing with version 2 of this object
else:
    # code dealing with version 1 of this object
```
I think readability might be improved by clearly using a `version` field to decide what to do.
```
if result['version'] = '1':
    # code for v 1
else:
    # code for v 2
```

So I'd propose to add a "version" field to `result` for the only purpose of improving readability on the consumer side.

Reply:
This is completely opposite to what we did before and would be a major API change in how we encourage usage of this module.

Up to now, we basically encouraged users (i.e. ctapipe) not to now anything about versions and use `.get` with leaving it up to them how they want to fallback if the value is not there.

I think this is nothing for this PR and not even for version 1. We could think about straightening our return values for a version 2. Let's move this discussion to an issue.

Reply:
I also think this approach is still better. Users of this module shouldn't have to care about versions, just about if the information they need is there or not.

Reply:
Okay. I forgot.